specification/machinelearning/Azure.AI.ChatProtocol/models/common.tsp

```diff
@@ -0,0 +1,60 @@
+import "@typespec/http";
+
+namespace Azure.AI.ChatProtocol;
+
+using TypeSpec.Http;
+
+@doc("Representation of the reason why a chat session has finished processing.")
+enum FinishReason {
+  @doc("Completion ended normally.")
+  stopped: "stop",
+
+  @doc("The completion exhausted available tokens before generation could complete.")
+  tokenLimitReached: "length",
+}
+
+@doc("A representation of the intended purpose of a message.")
+enum ChatRole {
+  @doc("The role that provides input to the completion.")
+  user: "user",
+
+  @doc("The role that instructs or configures the behavior of the assistant.")
+  system: "system",
+
+  @doc("The role that provides responses to the system-instructed, user-prompted input.")
+  assistant: "assistant",
+}
+
+@doc("A property that represents backend-specific context or arguments.")
+model ContextProperty<ContextModel> {
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Protocol defines the type as Record<any>"
+  @doc("""
+  Context allows the chat app to receive extra parameters from the client, such as temperature, functions, or
+  customer_info. These parameters are specific to the chat app and not understood by the generic clients.
+  """)
+  context?: ContextModel;
```

Comment:
Why refer to it as `ContextModel` and not just `Context`?

Reply:
no good reason, just to make it clearer (as it is a bad name) that is a template parameter.

Reply:
talked offline and updated to `TContext` and `TState`